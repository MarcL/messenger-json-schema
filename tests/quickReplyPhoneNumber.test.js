const ajv = require('./ajv');
const quickReplyPhoneNumber = require('../messenger/quickReplyPhoneNumber');

describe('Quick reply: phone number message', () => {
    let validate;
    
    beforeEach(() => {
        validate = ajv.compile(quickReplyPhoneNumber);
    });

    test('Valid message', () => {
        const givenMessage = {
            content_type: 'user_phone_number'
        };

        validate(givenMessage);
        
        expect(validate.errors).toBeNull();
    });

    describe('Invalid message', () => {
        const invalidScenarios = [
            {
                testMessage: 'Message is not a string',
                givenMessage: {
                    content_type: 100
                }
            },
            {
                testMessage: 'Message missing content_type property',
                givenMessage: {}
            },
            {
                testMessage: 'Message contains additional properties',
                givenMessage: {
                    content_type: 'user_phone_number',
                    invalid: 'Invalid property'
                }
            },
            {
                testMessage: 'Message contains invalid content_type property',
                givenMessage: {
                    content_type: 'not_user_email'
                }
            }
        ];

        invalidScenarios.forEach(scenario => {
            const {testMessage, givenMessage} = scenario;

            test(testMessage, () => {        
                validate(givenMessage);

                expect(validate.errors).toMatchSnapshot();
            });
        });
    });
});
