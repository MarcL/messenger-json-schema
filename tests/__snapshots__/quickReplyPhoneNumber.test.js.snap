// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quick reply: phone number message Invalid message Message contains additional properties 1`] = `
Array [
  Object {
    "dataPath": "",
    "keyword": "additionalProperties",
    "message": "should NOT have additional properties",
    "params": Object {
      "additionalProperty": "invalid",
    },
    "schemaPath": "#/additionalProperties",
  },
]
`;

exports[`Quick reply: phone number message Invalid message Message contains invalid content_type property 1`] = `
Array [
  Object {
    "dataPath": "/content_type",
    "keyword": "enum",
    "message": "should be equal to one of the allowed values",
    "params": Object {
      "allowedValues": Array [
        "user_phone_number",
      ],
    },
    "schemaPath": "#/properties/content_type/enum",
  },
]
`;

exports[`Quick reply: phone number message Invalid message Message is not a string 1`] = `
Array [
  Object {
    "dataPath": "/content_type",
    "keyword": "type",
    "message": "should be string",
    "params": Object {
      "type": "string",
    },
    "schemaPath": "#/properties/content_type/type",
  },
  Object {
    "dataPath": "/content_type",
    "keyword": "enum",
    "message": "should be equal to one of the allowed values",
    "params": Object {
      "allowedValues": Array [
        "user_phone_number",
      ],
    },
    "schemaPath": "#/properties/content_type/enum",
  },
]
`;

exports[`Quick reply: phone number message Invalid message Message missing content_type property 1`] = `
Array [
  Object {
    "dataPath": "",
    "keyword": "required",
    "message": "should have required property 'content_type'",
    "params": Object {
      "missingProperty": "content_type",
    },
    "schemaPath": "#/required",
  },
]
`;
